### ISSUE TITLE
Cannot get a value from /public/* of account

### ISSUE THREAD

**Issue by UCT10** [onflow/cadence#667]
@joshuahannan @turbolent 
I'm trying to figure out how to retrieve my Dapper Balance from blockchain. Unfortunately, none of the instructions work. See:
https://github.com/onflow/flow-ft/blob/1bef2c5862cb3e34753967389036efd53ac6776d/transactions/scripts/get_balance.cdc (this instruction actually helped)
https://github.com/onflow/flow/blob/6c5abc7e80c8fa5b39c4e5d79ab4e59f11d512ff/docs/content/cadence/tutorial/03-fungible-tokens.mdx#check-account-balances

And the way like in this one project doesn't work too. 
https://github.com/onflow/OWBSummer2020Project/blob/02f1e51b63eb7447935520a1f700118144c678e3/14/scripts/Script1.cdc

I'm using flow-cli with Linux terminal, with the following command:
`flow scripts execute transactions/scripts/get_totalSupply.cdc --host access.mainnet.nodes.onflow.org:9000`
With modified get_totalSupply.cdc respectively.

With this one code:
```import FungibleToken from 0xf233dcee88fe0abe
import DapperUtilityCoin from 0xead892083b3e2c6c
pub fun main(): Capability {
    let acct = getAccount(0x74040f9a566af82b)
    let ownerCapability = acct.getCapability(/public/dapperUtilityCoinReceiver)!
    return ownerCapability
}
```

I get this:
```
Failed to submit executable script: client: rpc error: code = Unknown desc = 2 errors occurred:
	* rpc error: code = Internal desc = failed to execute the script on the execution node execution-4ab025ab974e7ad7f344fbd16e5fbcb17fb8769fc8849b9d241ae518787695bd@execution-003.mainnet5.nodes.onflow.org:3569=1000: rpc error: code = DeadlineExceeded desc = context deadline exceeded
	* rpc error: code = Internal desc = failed to execute the script on the execution node execution-0ca407c1da940952ebcc02283b60cd97c9a008e111a48ea6cf1ce8f36f1e0153@execution-004.mainnet5.nodes.onflow.org:3569=1000: rpc error: code = DeadlineExceeded desc = context deadline exceeded
```

And so with others method. Now I'm getting error 

```
Failed to submit executable script: client: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp 35.226.224.190:9000: connect: connection refused"
```
Which should mean some network issues.

So, what am I doing wrong? Can I actually get Dapper balance from my account with blockchain? If you want to check, my TS Flow account is 0x74040f9a566af82b and my current balance is 112.85 USD. Thanks.

**Comment by UCT10** [comment]
Oh, and `return vaultRef.balance` doesn't work too, because there's no balance

**Comment by UCT10** [comment]
Hm, somehow I managed to get something with this one:
```
import FungibleToken from 0xf233dcee88fe0abe
import DapperUtilityCoin from 0xead892083b3e2c6c

pub fun main(): &DapperUtilityCoin.Vault{FungibleToken.Receiver}? {
    let acct = getAccount(0x74040f9a566af82b)
    let vaultRef = acct.getCapability(/public/dapperUtilityCoinReceiver).borrow<&DapperUtilityCoin.Vault{FungibleToken.Receiver}>()

    return vaultRef
}
```
The response is:
```
{"type":"Optional","value":{"type":"Resource","value":{"id":"A.e544175ee0461c4b.TokenForwarding.Forwarder","fields":[{"name":"uuid","value":{"type":"UInt64","value":"3863274"}},{"name":"recipient","value":{"type":"Capability","value":{"path":{"type":"Path","value":{"domain":"public","identifier":"dapperUtilityCoinReceiver"}},"address":"0xead892083b3e2c6c","borrowType":""}}}]}}}
```
It looks like 3863274 is the uuid in 0xead892083b3e2c6c's DapperUtilityCoin.Vault or something like this. Can I get amount of DUC from there? 
Or maybe there's a different approach, what's a transaction of deposit funds to Dapper from crypto or debit card? I didn't see anything of this type in blockchain. Can please you provide the txid of any?

**Comment by UCT10** [comment]
I'm a total newbie to cadence and flow blockchain, so forgive me. [It helped, but only particularly](https://github.com/onflow/flow-ft/blob/1bef2c5862cb3e34753967389036efd53ac6776d/transactions/scripts/get_balance.cdc)
But the result is frustrating. I cannot get any value except the Main Vault.

```
import FungibleToken from 0xf233dcee88fe0abe
import DapperUtilityCoin from 0xead892083b3e2c6c

pub fun main(): &DapperUtilityCoin.Vault{FungibleToken.Balance}? {
    let acct = getAccount(0xead892083b3e2c6c)
    let vaultRef = acct.getCapability(/public/dapperUtilityCoinBalance).borrow<&DapperUtilityCoin.Vault{FungibleToken.Balance}>()

    return vaultRef
}
```
The response is:
```
{"type":"Optional","value":{"type":"Resource","value":{"id":"A.ead892083b3e2c6c.DapperUtilityCoin.Vault","fields":[{"name":"uuid","value":{"type":"UInt64","value":"2"}},{"name":"balance","value":{"type":"UFix64","value":"999999999.00000000"}}]}}}
```
It seems that DUC was just minted at the beginning of the whole story, and the community might not be able to control the legitimate of its emission. And this is without a mention that Dapper users don't control either private keys, their money (DUC), only NFT collections and only through nbatopshot.com UI. If I'm wrong at anything, please, correct me.

Still, there's a question. Is there any way I can see deposits and withdrawals of TS user's money? Is there any way I can audit whether DUCs are backed by real dollars or not? If there's no such way, then why?
It's stated that Flow blockchain is transparent, so this feature is crucially required if it's not implemented yet.

**Comment by turbolent** [comment]
Hi there!

This is the repository of the Cadence programming language. Top Shot and the Dapper Wallet are products of Dapper Labs. 
We can't help with questions regarding specific smart contracts. Please consider asking the developers about their smart contracts. The best places for questions regarding Top Shot and Dapper is probably the [Top Shot Discord server](https://discord.com/invite/nbatopshot) and [the GitHub repository of the Top Shot contracts](https://github.com/dapperlabs/nba-smart-contracts).

Regarding the failure in your initial question: Your script was correct, but it actually uncovered a bug in Cadence, where returning capabilities without a type arguments was not working correctly. This short-coming was fixed in PR #666. 

> 3863274 is the uuid in 0xead892083b3e2c6c's DapperUtilityCoin.Vault

All resources in Cadence have a UUID. You can learn more about this feature in [the language reference](https://docs.onflow.org/cadence/language/composite-types#resource-identifier).

**Comment by UCT10** [comment]
Ok, thank you. I've discovered a few bugs for the first, so I've opened an issue in this repository.
